tempodb/encoding/vparquet/create.go

```diff
@@ -139,19 +132,29 @@ func (b *streamingBlock) AddRaw(id []byte, row parquet.Row, start, end uint32) e
 	b.bloom.Add(id)
 	b.meta.ObjectAdded(id, start, end)
 	b.currentBufferedTraces++
-	b.currentBufferedValues += len(row)
+	b.currentBufferedBytes += estimateProtoSize(row)
+
 	return nil
 }
 
-func (b *streamingBlock) CurrentBufferedValues() int {
-	return b.currentBufferedValues
+func (b *streamingBlock) EstimatedBufferedBytes() int {
+	return b.currentBufferedBytes
 }
 
 func (b *streamingBlock) CurrentBufferedObjects() int {
 	return b.currentBufferedTraces
 }
 
 func (b *streamingBlock) Flush() (int, error) {
+	// batch write traces
+	if len(b.bufferedTraces) > 0 {
+		_, err := b.pw.Write(b.bufferedTraces)
```

Comment:
It's very interesting that this results in ~40% less memory.  I would have expected an increase, since at this point the entire page of traces is resident in memory twice:  `b.bufferedTraces` and within `b.pw`.   Do you think it's worth zeroing out `b.bufferedTraces` to allow GC?  Otherwise it seems possible for Traces to hang around until the streaming block is collected.

Reply:
The memory savings come from simply exerting way less pressure on the GC. Agree we should zero out traces. I will add that.